pkg/cmd/kafka/consumergroup/resetoffset/reset_offset.go

```diff
@@ -0,0 +1,246 @@
+
+	offsetResetParams := kafkainstanceclient.ConsumerGroupResetOffsetParameters{
+		Offset: opts.offset,
+	}
+
+	if opts.value != "" {
+		offsetResetParams.Value = &opts.value
+	}
+
+	if opts.topic != "" {
+		topicToReset := kafkainstanceclient.TopicsToResetOffset{
+			Topic: opts.topic,
+		}
+
+		if len(opts.partitions) != 0 {
+			topicToReset.Partitions = &opts.partitions
+		}
+
+		topicsToResetArr := []kafkainstanceclient.TopicsToResetOffset{topicToReset}
+
+		offsetResetParams.Topics = &topicsToResetArr
+	}
+
+	a := api.GroupsApi.ResetConsumerGroupOffset(ctx, opts.id).ConsumerGroupResetOffsetParameters(offsetResetParams)
+
+	if !opts.skipConfirm {
+
+		var confirmReset bool
+		opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.input.confirmReset.message", localize.NewEntry("ID", opts.id))
+		promptConfirmReset := &survey.Confirm{
+			Message: opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.input.confirmReset.message", localize.NewEntry("ID", opts.id)),
+		}
+
+		if err = survey.AskOne(promptConfirmReset, &confirmReset); err != nil {
+			return err
+		}
+		if !confirmReset {
+			logger.Debug(opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.log.debug.cancelledReset"))
+			return nil
+		}
+	}
+
+	updatedConsumers, httpRes, err := a.Execute()
+
+	if err != nil {
+
+		if httpRes == nil {
+			return err
+		}
+
+		operationTmplPair := localize.NewEntry("Operation", "reset offset")
+
+		switch httpRes.StatusCode {
+		case http.StatusUnauthorized:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.unauthorized", operationTmplPair))
+		case http.StatusForbidden:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.forbidden", operationTmplPair))
+		case http.StatusInternalServerError:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.internalServerError"))
+		case http.StatusServiceUnavailable:
+			return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.unableToConnectToKafka", localize.NewEntry("Name", kafkaInstance.GetName())))
+		default:
+			return err
+		}
+	}
+
+	defer httpRes.Body.Close()
+
+	logger.Info(opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.log.info.successful", localize.NewEntry("ConsumerGroupID", opts.id), localize.NewEntry("InstanceName", kafkaInstance.GetName())))
+
+	switch opts.output {
+	case "json":
+		data, _ := json.Marshal(updatedConsumers)
+		_ = dump.JSON(opts.IO.Out, data)
+	case "yaml", "yml":
+		data, _ := yaml.Marshal(updatedConsumers)
+		_ = dump.YAML(opts.IO.Out, data)
+	default:
+		logger.Info("")
+		consumers := updatedConsumers.GetItems()
+		rows := mapResetOffsetResultToTableFormat(consumers)
+		dump.Table(opts.IO.Out, rows)
+
+		return nil
+	}
+
+	return nil
+
+}
+
+func mapResetOffsetResultToTableFormat(consumers []kafkainstanceclient.ConsumerGroupResetOffsetResultItem) []UpdatedConsumerRow {
+	rows := []UpdatedConsumerRow{}
```

Comment:
```suggestion
	rows := make([]UpdatedConsumerRow, len(consumers))
```

Since we know what the length of `rows` will be, we could create a fixed sized array. This will save some memory. It's not going to make a massive difference at all and it's fine to leave it as is, but a good practice to adopt :)

I'm aware we have not been doing this in other places.

Reply:
Using make is throwing lint error at append:

`append to slice `rows` with non-zero initialized length (makezero)`

Reply:
Since it would create a slice with a fixed size of `n`, if you append to it you would be adding the new value to position `n+1`. You need to iterate over `rows` and assign each value using the index.